packages/plugins/plugin-space/src/components/SyncStatus.tsx

```diff
@@ -0,0 +1,136 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import { CloudArrowDown, CloudArrowUp, CloudCheck, CloudSlash } from '@phosphor-icons/react';
+import React, { useEffect } from 'react';
+
+import type { UnsubscribeCallback } from '@dxos/async';
+import type { Client } from '@dxos/client';
+import { type Space, type SpaceId, type SpaceSyncState } from '@dxos/client/echo';
+import { Context } from '@dxos/context';
+import { StatusBar } from '@dxos/plugin-status-bar';
+import { EdgeService } from '@dxos/protocols';
+import { useClient } from '@dxos/react-client';
+import { getSize } from '@dxos/react-ui-theme';
+
+export const SyncStatus = () => {
+  const client = useClient();
+  const [state, setState] = React.useState<EdgeSyncState>(initialEdgeSyncState);
+
+  useEffect(() => {
+    return createClientSyncTracker(client, (state) => {
+      setState(state);
+    });
+  }, []);
+
+  const needsToUpload = state.missingOnRemote > 0 || state.differentDocuments > 0;
+  const needsToDownload = state.missingOnLocal > 0 || state.differentDocuments > 0;
+  const notConnectedToEdge = state.spacesNotConnectedToEdge.size > 0;
+
+  return (
+    <StatusBar.Item title={JSON.stringify(state, null, 2)}>
+      {notConnectedToEdge ? (
+        <CloudSlash className={getSize(4)} />
+      ) : needsToUpload ? (
+        <CloudArrowUp className={getSize(4)} />
+      ) : needsToDownload ? (
+        <CloudArrowDown className={getSize(4)} />
+      ) : (
+        <CloudCheck className={getSize(4)} />
+      )}
+    </StatusBar.Item>
+  );
+};
+
+const createClientSyncTracker = (client: Client, cb: (state: EdgeSyncState) => void) => {
+  const unsubscribeCallbacks: Record<SpaceId, UnsubscribeCallback> = {};
+  const state: Record<SpaceId, SpaceSyncState> = {};
+
+  const install = (spaces: Space[]) => {
+    for (const space of spaces) {
+      if (state[space.id]) {
+        continue;
+      }
+
+      unsubscribeCallbacks[space.id] = space.crud.subscribeToSyncState(Context.default(), (syncState) => {
+        state[space.id] = syncState;
+        cb(extractEdgeSyncSate(state));
+      });
+    }
+  };
+  client.spaces.subscribe((spaces) => {
+    install(spaces);
+  });
+  install(client.spaces.get());
+
+  return () => {
+    for (const unsubscribe of Object.values(unsubscribeCallbacks)) {
+      unsubscribe();
+    }
+  };
+};
+
+type EdgeSyncState = {
```

Comment:
is there a better place to put all of the edge deps? (outside of ux?)

Reply:
so its not in plugin-space?

yeah, probably we should make an edge plugin